tensorrt_llm/_torch/auto_deploy/transformations/library/sharding.py

```diff
@@ -56,6 +67,8 @@ class ShardingTransformInfo(BaseModel, ABC):
     target_node: str
     rank: int
     world_size: int
+    dist_backend: DistBackend = DistBackend.AUTO
+    trtllm_allreduce_strategy: AllReduceStrategy = AllReduceStrategy.AUTO
```

Comment:
```suggestion
    trtllm_allreduce_strategy: int = 0
```

@suyoggupta it would just be like this then and we would just convert it to the `AllReduceStrategy` enum inside the trtllm-specific custom op?